### DISCUSSION TITLE
Welcome to Win11Debloat Discussions!

### DISCUSSION THREAD

**Discussion by Raphire** [Raphire/Win11Debloat#73] [Announcements]
## Welcome!
In general, the intent of Discussions is to be a dedicated space for both users and collaborators for asking general questions and to discuss features, share ideas, and provide support for Win11Debloat.  

If you encounter any bugs or would like to request a new feature, please [create an issue](https://github.com/Raphire/Win11Debloat/issues) instead.

**Comment by IcculusOfTheSky** [comment]
Amazing script, got my pc cpu idle down a bit, thanks!

**Reply by greatquux** [reply]
lol even the author of the helping friendly book needs to get rid of Microsoft’s crap!

**Comment by cjcombsMedia** [comment]
So what is the reality of debloating Win11 Pro to NOT require online account once a Win11 instance is already installed? Is that outside of the scope of what Win11Debloat does?

**Reply by Raphire** [reply]
For most users I don't see a good way to integrate this into Win11Debloat as the script is intended to be run on an existing user. 

Win11Debloat does have a Sysprep mode which works before account creation. It could possibly be added as an option there, but Sysprep mode is a niche feature aimed at system admins. Most people will not be running scripts (or even know how to do this) before account creation.

I think for the 'regular' person tools like rufus are a much better option to disable the network/online account requirement.

**Comment by Revermb** [comment]
Does the automated PS script version auto delete itself after it's done debloating or is it saved in PS as a module?
Talking about this one: `& ([scriptblock]::Create((Invoke-RestMethod "https://win11debloat.raphi.re/")))`

**Reply by Raphire** [reply]
All traces are deleted after the script is done executing, as long as you don't close the initial PS window before the script is done. 

You can see the code for it here, it's quite simple: https://github.com/Raphire/Win11Debloat/blob/master/Get.ps1

It downloads the script as a zip from github, extracts it, removes the zip file, runs the script and deletes the remaining files once the script is done. Everything is saved to the users temp directory, which I believe is at `C:\Users\{USERNAME}\AppData\Local\Temp`.

**Reply by Revermb** [reply]
Alright thanks. Appreciate the reply.

**Comment by rdelpellegrino** [comment]
I am reluctant to raise a bug report on something that is likely to be a false positive. However, after the recent update:
2024-09 Cumulative Update for Windows 11 Version 23H2 for x64-based Systems (KB5043076)

I re-ran the script. It it was blocked by Bitdefender AV. 

pwsh> & ([scriptblock]::Create((irm "https://win11debloat.raphi.re/")))
At line:1 char:1
+ param (
+ ~~~~~~~
This script contains malicious content and has been blocked by your antivirus software.
    + CategoryInfo          : ParserError: (:) [], ParseException
    + FullyQualifiedErrorId : ScriptContainedMaliciousContent
  
  Bitdefender reports:  
  Suspicious activity blocked
12 minutes ago
Feature:
Antivirus
PowerShell tried to load a malicious resource detected as Heur.BZC.ZFV.Boxter.341.FE610AB7 and was blocked. Your device is safe.

This did not occur when I ran the script some weeks ago. The correlation with the Windows update is suspicious. It is not clear to me where this is being triggered as the relationship between AV products and Windows is something I rather not dwell on...

Cannot rule out that this is just an issue on my machine. So, it might be helpful if someone else can reproduce this.

**Reply by Raphire** [reply]
Interesting. Thanks for reporting this.

In my quick search I wasn't able to find any references to `Heur.BZC.ZFV.Boxter.341.FE610AB7`, so it's hard to say what's causing this specific detection. Although my best guess is that it's related to this command downloading the script from github and executing it. That is admitedly something that can be consideren suspicious if it happens when the user doesn't have knowledge of this.

**Reply by rdelpellegrino** [reply]
I think you are correct. I downloaded the archive from Github and ran it locally with Admin privileges and it worked just fine.
TBH. I have more confidence in your code than Microsoft's - which is why I am using this tool after all.
FWIW I think users can safely ignore this message. If the error appears in the script download option, just download the archive it and run it locally.
 Microsoft have been aggressively forcing unwanted spyware onto user's machines and I believe are likely to take actions against tools such as this that try to prevent it. I will raise this matter with Bitdefender who, after all, I pay to defend me from the likes of Microsoft. If their product is going to work against that then time to look elsewhere.

**Comment by LoafusCramwell** [comment]
Have you received any complaints about the System Tray breaking after running the Debloater? I'd love to use it in my organization, but every computer i run it on ends up breaking the system Tray. When clicked on (the network/sound/battery area), the tray appears momentarily, then disappears.  I've ruled out other build conditions (like group policies). 

I don't know if this is the result of the Debloater uninstalling an app, or if it's some of the other options (file explorer, task bar, etc.). Any help would be appreciated!

**Reply by Raphire** [reply]
Heya,

I can't say I've run into this or heard of anyone running into this. I can't really explain it either as the script doesn't really touch anything that could logically cause it to break. Does this behaviour stay after a restart?

**Reply by LoafusCramwell** [reply]
Thanks for the response! I ended up running each facet of the script incrementally and then rebooting (First app removal, then start menu, task bar, etc etc.). I ended up not being able to replicate the issue. 

Turns out the issue was with a separate script I ran right after, which disables a bunch of services we don't want enabled in our org. We carried the script over from our Windows 10 build procedure:
- If you DISABLE the "Wi-Fi Direct Services Connection Manager Service" and then reboot, the System Tray / Action Center breaks! As soon as it's enabled again, the ST/AC works without issue.

Sorry to jump to the conclusion that it was your script. I hope this helps somebody in the future.

**Reply by Raphire** [reply]
Thanks for reporting back, glad you found the issue :)

**Comment by West1483** [comment]
Nice

**Comment by magnumquest39-cloud** [comment]
@magnumquest39@gmail.com
